src/lib.rs

```diff
@@ -168,6 +172,24 @@ pub fn parse_datetime_at_date<S: AsRef<str> + Clone>(
         }
     }
 
+    // parse weekday
+    if let Ok(weekday) = parse_weekday::parse_weekday(s.as_ref()) {
+        let mut beginning_of_day = Local
+            .with_ymd_and_hms(date.year(), date.month(), date.day(), 0, 0, 0)
+            .unwrap();
+
+        loop {
+            if beginning_of_day.weekday() == weekday {
+                break;
+            }
+            beginning_of_day += Duration::days(1);
+        }
```

Comment:
You've written your own `while` loop :)
```suggestion
        while beginning_of_day.weekday() != weekday {
            beginning_of_day += Duration::days(1);
        }
```